sympy/printing/smtlib.py

```diff
@@ -168,11 +169,26 @@ def _print_BooleanFalse(self, x: BooleanFalse):
         return 'false'
 
     def _print_Float(self, x: Float):
-        f = x.evalf(self._precision) if self._precision else x.evalf()
-        return str(f).rstrip('0')
+        x = x.evalf(self._precision) if self._precision else x.evalf()
+        dps = prec_to_dps(x._prec)
+        str_real = mlib_to_str(x._mpf_, dps, strip_zeros=True, min_fixed=None, max_fixed=None)
+
+        if 'e' in str_real:
+            (mant, exp) = str_real.split('e')
+
+            if exp[0] == '+':
+                exp = exp[1:]
+
+            return r"(* %s (^ 10 %s))" % (mant, exp)
```

Comment:
this should use whatever power and mult symbols are specified in the symbols rather than using ^ and *